tools/mason/MasonSystem.chpl

```diff
@@ -19,34 +19,46 @@
  */
 
 
-private use List;
-private use Map;
-use Path;
-use MasonUtils;
+use List;
+use Map;
+use MasonArgParse;
 use MasonHelp;
+use MasonUtils;
+use Path;
 use Regex;
 
 /* Entry point for mason system commands */
 proc masonSystem(args: [] string) {
+  var parser = new argumentParser();
+
+  var helpFlag = parser.addFlag("help",
+                                opts=["-h","--help"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var pcCmd = parser.addSubCommand("pc");
+  var searchCmd = parser.addSubCommand("search");
   try! {
-    if args.size < 3 {
-      masonSystemHelp();
-      exit(0);
-    }
-    else if args[2] == "--help" || args[2] == "-h" {
-      masonSystemHelp();
-      exit(0);
-    }
-    else if pkgConfigExists() {
-      if args[2] == "pc" {
-        printPkgPc(args);
-      }
-      else if args[2] == "search" {
-        pkgSearch(args);
+    parser.parseArgs(args);
+  }
+  catch ex : ArgumentError {
+    stderr.writeln(ex.message());
+    masonSystemHelp();
+    exit(1);
+  }
+  if helpFlag.valueAsBool() {
+    masonSystemHelp();
+    exit(0);
+  }
+  try! {
+
+    if pkgConfigExists() {
```

Comment:
I think we should swap these conditionals, so it is like this

```
if pcCmd.hasValue() {
  if pkgConfigExists() {
  }
} else if searchCmd.hasValue() {
  if pkgConfigExists() {
  }
}
```

Then, in this PR or in a future one, we should modify it to give an error about not finding pkg-config rather than a couldn't parse arguments error or failing silently.

Reply:
In addition to re-ordering the logic, since `pkgConfigExists()` never returns false, what if we drop the `if` because it will either be there or it will throw an exception with the message that pkg-config wasn't found.

Reply:
sounds good